Makefile

```diff
@@ -14,7 +14,8 @@ install:
 
 lint:
 	go vet -all -printfuncs=Verbosef,Infof,Debugf ./...
-
+    ! grep --include=\*.go -rn . -e '^[^/].*defer [^ ]*EndPipeline(' # linting for improperly deferred EndPipeline calls; should be in closure, i.e. `defer func() { ...EndPipeline(err) }()`
```

Comment:
i'm not sure this is a good idea, this seems potentially buggy and papering over a bad api